### DISCUSSION TITLE
PR limit blocked other possible auto-mergable updates

### DISCUSSION THREAD

**Discussion by ckychris** [renovatebot/renovate#19501] [Suggest an Idea]
We are using the auto-merge feature to make updates as automatic as possible.

We have a low PR/branch limit (3/5). However, this gets the whole renovate process stuck if open PRs are not resolved. 

I am aware of `dependencyDashboardApproval`, which helps with this problem. The potential change we can make is to add all major updates to `dependencyDashboardApproval`.

### Will a new config feature help?
`automerge-cycling-limit`: `1`

If automerge is true for some packages, create x number of branches and delete those if they don't pass static checks.

e.g. 3 packages that cannot be updated, 3 PRs created that don't pass static checks. The PR limit is reached. Because of `automerge-cycling-limit`, renovate will open a new branch to attempt other dependency updates first. If the new branch fails to pass static checks, it will be closed, and another branch containing other dependency updates will be opened.

The ultimate goal is to make renovate able to automerge all possible updates with a set branch/PR limit.

### Related questions
Renovatebot dealing with PRs open more than X period of time
https://github.com/renovatebot/renovate/discussions/11804

This proposal is different from removing stale PRs. Stale PRs will continue to open and be ready to resolve.

**Comment by rarkins** [comment]
I would like to avoid a new feature for this. If you have non passing PRs then close them to ignore/defer and that would allow new PRs to be created.

Also, you can set a higher `branchConcurrentLimit` than `prConcurrentLimit`, and you can also set `prCreation=status-success` so that PRs are only created if tests pass. It will mean that your failing branches are left in a sort of "limbo" (branch exists but PR never created) but you will get most of what you're after.
